internal/app/handlers.go

```diff
@@ -74,6 +77,58 @@ func (app *Application) getCodeSystemByID(w http.ResponseWriter, r *http.Request
 	json.NewEncoder(w).Encode(codeSystem)
 }
 
+func (app *Application) getFHIRCodeSystemByID(w http.ResponseWriter, r *http.Request) {
+	rp := app.repository
+
+	id := r.PathValue("id")
+	id_type, err := determineIdType(id)
+	if err != nil {
+		customErrors.BadRequest(w, r, err, app.logger)
+		return
+	}
+
+	var codeSystem *xo.CodeSystem
+	if id_type == "oid" {
+		codeSystem, err = rp.GetCodeSystemByOID(r.Context(), id)
+	} else {
+		codeSystem, err = rp.GetCodeSystemByID(r.Context(), id)
+	}
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			errorString := fmt.Sprintf("Error: Code System %s not found", id)
+			dbErr := &customErrors.DatabaseError{
+				Err:    err,
+				Msg:    errorString,
+				Method: "getCodeSystemById",
+				Id:     id,
+			}
+			dbErr.NoRows(w, r, err, app.logger)
+		} else {
+			customErrors.ServerError(w, r, err, app.logger)
+		}
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	fhirCodeSystem, err := r5.SerializeCodeSystemToFhir(codeSystem)
+	if err != nil {
+		customErrors.ServerError(w, r, err, app.logger)
+	}
+
+	marshaller, err := jsonformat.NewMarshaller(false, "", "", fhirversion.R4)
```

Comment:
Any idea what `fhirversion.R4` does here? I was expecting `fhirversion.R5` but I see that's not an option available to us 🤔

Reply:
Yeah, they don't support R5 yet for the Marshaller. I think from what I can tell though, this just handles formatting as it converts to JSON, so I think it still works for our purposes. This might become an issue if we ever need to Unmarshal JSON to the Go types.